src/renderers/dom/shared/devtools/ReactDOMUnknownComponentDevtool.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactDOMUnknownComponentDevtool
+ */
+
+'use strict';
+
+var ReactComponentTreeDevtool = require('ReactComponentTreeDevtool');
+var warning = require('warning');
+var ReactDOMFactories = require('ReactDOMFactories');
+var didWarnAboutCase = false;
+
+function handleElement(debugID, element) {
+  if (element == null || typeof element.type !== 'string' ||
+    element.type.match(/\-|\s+/) !== null ||
+    ReactDOMFactories.hasOwnProperty(element.type)) {
```

Comment:
@Aweary completely agree with you, but other than keeping a list of elements, how else can we make sure there's no false positives?

According to the the [specs](http://w3c.github.io/html-reference/syntax.html#svg-mathml), SVG and MathML are elements allowed in HTML which belong to their own respective namespace, so createElement will return `HTMLUnknownElement` for any of these elements, aside from those, everything else should represent `HTMLUnknownElement` accurately.

Seems to me the solution for now is, we have to list svg/mathml valid tags and make sure we "ignore" those as valid. 

In a perfect world we could get this data from W3C but I couldn't find anything useful to consume other than scraping their repo or MDN. They do have the [useful tests](https://github.com/w3c/web-platform-tests) that we can probably gather something from.

However this is essentially keeping a list of elements (kind of like the same ReactDOMFactories is doing) which it was already said is not the best solution. What do you guys think would be the ideal solution?